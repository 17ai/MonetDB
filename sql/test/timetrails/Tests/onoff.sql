-- on-off query problem. Find all the values that form a consecutive list

CREATE TABLE coil( tick integer, level integer);

INSERT INTO coil
VALUES
(1,0),(2,0),(3,0),(4,0),(5,1),(6,1),(7,1),(8,0),(9,2),(10,2),(11,2),(12,2),(13,0),(14,0),(15,3),(16,0),(17,0),(18,0),(19,4),(20,4),(21,4),(22,5),(23,5),(24,5),(25,0),(26,0),(27,6),(28,0),(29,6),(30,0),(31,7),(32,7),(33,7),(34,7),(35,8),(36,8),(37,8),(38,8),(39,8),(40,7),(41,7),(42,7),(43,7),(44,9),(45,9),(46,9),(47,0),(48,0),(49,0);

select A.level, count(*)from coil A, coil B where A.tick+1 = B.tick and A.level <> B.level group by A.level;
