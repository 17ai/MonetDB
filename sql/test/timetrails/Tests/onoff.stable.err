stderr of test 'onoff` in directory 'sql/test/timetrails` itself:


# 20:56:51 >  
# 20:56:51 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=31197" "--set" "mapi_usock=/var/tmp/mtest-14521/.s.monetdb.31197" "--set" "monet_prompt=" "--forcemito" "--dbpath=/export/scratch1/mk/timetrails//Linux/var/MonetDB/mTests_sql_test_timetrails"
# 20:56:51 >  

# builtin opt 	gdk_dbpath = /export/scratch1/mk/timetrails//Linux/var/monetdb5/dbfarm/demo
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_vmtrim = no
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 31197
# cmdline opt 	mapi_usock = /var/tmp/mtest-14521/.s.monetdb.31197
# cmdline opt 	monet_prompt = 
# cmdline opt 	gdk_dbpath = /export/scratch1/mk/timetrails//Linux/var/MonetDB/mTests_sql_test_timetrails
# cmdline opt 	gdk_debug = 536870922

# 20:56:51 >  
# 20:56:51 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-14521" "--port=31197"
# 20:56:51 >  


# 20:56:51 >  
# 20:56:51 >  "Done."
# 20:56:51 >  

