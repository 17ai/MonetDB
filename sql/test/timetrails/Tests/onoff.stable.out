stdout of test 'onoff` in directory 'sql/test/timetrails` itself:


# 20:56:51 >  
# 20:56:51 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=31197" "--set" "mapi_usock=/var/tmp/mtest-14521/.s.monetdb.31197" "--set" "monet_prompt=" "--forcemito" "--dbpath=/export/scratch1/mk/timetrails//Linux/var/MonetDB/mTests_sql_test_timetrails"
# 20:56:51 >  

# MonetDB 5 server v11.28.0
# This is an unreleased version
# Serving database 'mTests_sql_test_timetrails', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 128bit integers
# Found 15.589 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2017 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.da.cwi.nl:31197/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-14521/.s.monetdb.31197
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded

Ready.

# 20:56:51 >  
# 20:56:51 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-14521" "--port=31197"
# 20:56:51 >  

#CREATE TABLE coil( tick integer, level integer);
#INSERT INTO coil
#VALUES
#(1,0),(2,0),(3,0),(4,0),(5,1),(6,1),(7,1),(8,0),(9,2),(10,2),(11,2),(12,2),(13,0),(14,0),(15,3),(16,0),(17,0),(18,0),(19,4),(20,4),(21,4),(22,5),(23,5),(24,5),(25,0),(26,0),(27,6),(28,0),(29,6),(30,0),(31,7),(32,7),(33,7),(34,7),(35,8),(36,8),(37,8),(38,8),(39,8),(40,7),(41,7),(42,7),(43,7),(44,9),(45,9),(46,9),(47,0),(48,0),(49,0);
[ 49	]
#select A.level, count(*)from coil A, coil B where A.tick+1 = B.tick and A.level <> B.level group by A.level;
% sys.a,	sys.L4 # table_name
% level,	L3 # name
% int,	bigint # type
% 1,	1 # length
[ 0,	7	]
[ 1,	1	]
[ 2,	1	]
[ 3,	1	]
[ 4,	1	]
[ 5,	1	]
[ 6,	2	]
[ 7,	2	]
[ 8,	1	]
[ 9,	1	]

# 20:56:51 >  
# 20:56:51 >  "Done."
# 20:56:51 >  

